File: 1- Find pair that sums up to k/Java/solution3.java
// By using a dictionary:

// Time complexity: O(n)
// Space complexity: O(n)


static boolean findPair(int[] arr, int k){
  HashMap<Integer,Boolean> visited = new HashMap<Integer,Boolean>();
  for(int element : arr){
    if(visited.containsKey(k-element)) return true;
    else visited.put(element, true);
  }
  return false;
}

File: 1- Find pair that sums up to k/C++/solution3.cpp
// By using a dictionary:

// Time complexity: O(n)
// Space complexity: O(n)

bool findPair(int* arr, int arrLength, int k){
  unordered_map<int,bool> visited;
  for(int i = 0; i < arrLength; i++){
    if(visited.find(k-arr[i]) != visited.end()) return true;
    else visited[arr[i]] = true;
  }
  return false;
}


